core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -977,8 +977,7 @@ private void addInodeFromJournalInternal(Inode<?> inode) {
     mInodes.add(inode);
     // If journal entry has no security enabled, change the replayed inode permission to be 0777
     // for backwards-compatibility.
-    if (SecurityUtils.isSecurityEnabled() && inode != null && inode.getOwner().isEmpty()
-        && inode.getGroup().isEmpty()) {
+    if (SecurityUtils.isSecurityEnabled() && inode != null && !hasMode) {
```

Comment:
Can we move this `hasMode` type of check into the `fromJournalEntry` methods of `InodeFile` and `InodeDirectory`?